core/arch/aarch64/aarch64.asm

```diff
@@ -226,16 +226,42 @@ save_priv_mcontext_helper:
         str      w2, [x0, #(16 * ARG_SZ*2 + 12)]
         str      w3, [x0, #(16 * ARG_SZ*2 + 16)]
         add      x4, x0, #simd_OFFSET
-        st1      {v0.2d-v3.2d}, [x4], #64
-        st1      {v4.2d-v7.2d}, [x4], #64
-        st1      {v8.2d-v11.2d}, [x4], #64
-        st1      {v12.2d-v15.2d}, [x4], #64
-        st1      {v16.2d-v19.2d}, [x4], #64
-        st1      {v20.2d-v23.2d}, [x4], #64
-        st1      {v24.2d-v27.2d}, [x4], #64
-        st1      {v28.2d-v31.2d}, [x4], #64
-        /* TODO i#5365: Save Z/P regs as well? Will require runtime check of
-         * ID_AA64PFR0_EL1 for FEAT_SVE.
+
+        /* Registers Q0-Q31 map directly to registers V0-V31. */
+        str      q0, [x4], #64
```

Comment:
In addition to assuming the vec len is the full 64 bytes, was the ST1 also storing just one D (half-size element) instead of the whole V==Q?

Reply:
As I understand the previous version did vx.2d which means 2 64-bit floating point = 128 bits. And it stored 4 such together (==64 bytes) in one st1 op. @jackgallagher-arm to confirm.

Reply:
> As I understand the previous version did vx.2d which means 2 64-bit floating point = 128 bits. And it stored 4 such together (==64 bytes) in one st1 op. @jackgallagher-arm to confirm

That is correct.

Reply:
OK, so there are 2 variants of `ST1`: single-lane and multi-lane, all under the same `ST1` opcode.  But why call it `ST1` for multi-lane?  I thought the `1` meant "one lane".